flytekit/core/type_engine.py

```diff
@@ -487,7 +504,25 @@ def to_literal(cls, ctx: FlyteContext, python_val: typing.Any, python_type: Type
         transformer = cls.get_transformer(python_type)
         if transformer.type_assertions_enabled:
             transformer.assert_type(python_type, python_val)
+
+        # In case the value is an annotated type we inspect the annotations and look for hash-related annotations.
+        hash = None
+        if transformer.hash_overridable and get_origin(python_type) is Annotated:
```

Comment:
get_origin works in all 3.7-3.10?

Reply:
yes, it goes even further, it's 3.6 and newer.